pkg/reconciler/identity/reconciler_test.go

```diff
@@ -123,57 +102,76 @@ func TestCreates(t *testing.T) {
 				BlockOwnerDeletion: &trueVal,
 			}}),
 		),
-		expectedErr: fmt.Sprintf("adding iam policy binding failed with: " +
-			addIamPolicyBinding(context.Background(), projectID, gServiceAccountName, NewServiceAccount(kServiceAccountName, testNS, gServiceAccountName)).Error(),
-		),
+		wantErrCode: codes.NotFound,
 	}}
 
-	defer logtesting.ClearAll()
 	for _, tc := range testCases {
-		cs := fakeKubeClient.NewSimpleClientset(tc.objects...)
-		identity := &Identity{
-			KubeClient: cs,
-		}
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
 
-		arl := pkgtesting.ActionRecorderList{cs}
-		kserviceAccount, err := identity.ReconcileWorkloadIdentity(context.Background(), projectID, identifiable)
+			cs := fakeKubeClient.NewSimpleClientset(tc.objects...)
+			iam := gclient.NewTestClient()
+			m, err := NewIAMPolicyManager(ctx, iam)
+			if err != nil {
+				t.Fatal(err)
+			}
+			identity := &Identity{
+				KubeClient:    cs,
+				PolicyManager: m,
+			}
+			identifiable := NewCloudPubSubSource(identifiableName, testNS,
+				WithCloudPubSubSourceGCPServiceAccount(gServiceAccountName),
+				WithCloudPubSubSourceServiceAccountName("test"))
 
-		if (tc.expectedErr != "" && err == nil) ||
-			(tc.expectedErr == "" && err != nil) ||
-			(tc.expectedErr != "" && err != nil && tc.expectedErr != err.Error()) {
-			t.Errorf("Test case %q, Error mismatch, want: %q got: %q", tc.name, tc.expectedErr, err)
-		}
-		if diff := cmp.Diff(tc.expectedServiceAccount, kserviceAccount, ignoreLastTransitionTime); diff != "" {
-			t.Errorf("Test case %q, unexpected topic (-want, +got) = %v", tc.name, diff)
-		}
+			arl := pkgtesting.ActionRecorderList{cs}
+			kserviceAccount, err := identity.ReconcileWorkloadIdentity(context.Background(), projectID, identifiable)
```

Comment:
How about using `ctx` instead? So that we are sure everything gets cancelled eventually.

Reply:
Done, although I don't think this will call will leave anything running when it returns.